src/expected_input.rs

```diff
@@ -38,13 +38,23 @@ impl ExpectedInput {
         let mut string_vec: Vec<String> = str_vec.iter().map(|s| s.to_string()).collect();
         str_vec.shuffle(&mut rng);
 
+        // creating a pointer which points to where the words starts in the vector to help with uppercase words since we replace
+        // words in the beginning with numbers when numbers are enabled.
+        let mut words_start_pos:usize = 0;
+
         if config.numbers == true {
-            replace_words_with_numbers(&mut string_vec, &mut rng, config.numbers_ratio);
+            words_start_pos = replace_words_with_numbers(&mut string_vec, &mut rng, config.numbers_ratio);
+            str_vec = string_vec.iter().map(|s| s.as_str()).collect();
+        }
 
+        if config.uppercase == true {
+            let uppercase_string_vec = create_uppercase_words(&string_vec, words_start_pos, config.uppercase_ratio);
+            string_vec = uppercase_string_vec;
             str_vec = string_vec.iter().map(|s| s.as_str()).collect();
-            str_vec.shuffle(&mut rng);
         }
 
+        str_vec.shuffle(&mut rng);
+
```

Comment:
I thought about doing this but decided against it since we would need to maintain the ratio of uppercase to lowercase after we have added the numbers. If both flags are enabled then the addition of numbers would mess with the ratio.

Should I add additional comments to explain why I created pos better?

Reply:
I think the one that you added is enough 👍